geoips/interfaces/base.py

```diff
@@ -460,10 +411,56 @@ def get_plugin(self, name):
         ProductsInterface which uses a tuple containing 'source_name' and
         'name'.
         """
-        try:
-            validated = self.validator.validate(self._unvalidated_plugins[name])
-        except KeyError:
-            raise PluginError(f"Plugin '{name}' not found for '{self.name}' interface.")
+        from importlib.resources import files
+
+        if not self._unvalidated_plugins:
+            raise PluginRegistryError(
+                "Plugin registries not found, please run 'create_plugin_registries'"
+            )
+        if isinstance(name, tuple):
+            # These are stored in the yaml as str(name),
+            # ie "('viirs', 'Infrared')"
+            try:
+                relpath = self._unvalidated_plugins[self.name][name[0]][name[1]][
+                    "relpath"
+                ]
+                package = self._unvalidated_plugins[self.name][name[0]][name[1]][
+                    "package"
+                ]
+            except KeyError:
+                raise PluginError(
+                    f"Plugin [{name[1]}] doesn't exist under source name [{name[0]}]"
+                )
+            abspath = str(files(package) / relpath)
+            plugin = yaml.safe_load(open(abspath, "r"))
+            plugin_found = False
+            for product in plugin["spec"]["products"]:
+                if product["name"] == name[1] and name[0] in product["source_names"]:
+                    plugin_found = True
+                    plugin = product
+                    break
+            if not plugin_found:
+                raise PluginError(
+                    "There is no plugin that has " + name[1] + " included in it."
+                )
+            plugin["interface"] = "products"
+            plugin["package"] = package
+            plugin["abspath"] = abspath
+            plugin["relpath"] = relpath
+        else:
+            try:
+                relpath = self._unvalidated_plugins[self.name][name]["relpath"]
+                package = self._unvalidated_plugins[self.name][name]["package"]
+            except KeyError:
+                raise PluginError(
+                    f"Plugin [{name}] doesn't exist under interface [{self.name}]"
+                )
```

Comment:
Large try/except replaced with this and https://github.com/NRLMMD-GEOIPS/geoips/pull/240/files/#r1373527621

https://github.com/NRLMMD-GEOIPS/geoips/pull/240#discussion_r1370855696